grid-cli/cmd/deploy_vm.go

```diff
@@ -223,7 +209,116 @@ func init() {
 
 	deployVMCmd.Flags().Bool("ipv4", false, "assign public ipv4 for vm")
 	deployVMCmd.Flags().Bool("ipv6", false, "assign public ipv6 for vm")
-	deployVMCmd.Flags().Bool("ygg", true, "assign yggdrasil ip for vm")
+	deployVMCmd.Flags().Bool("ygg", false, "assign yggdrasil ip for vm")
 	deployVMCmd.Flags().Bool("mycelium", true, "assign mycelium ip for vm")
 	deployVMCmd.Flags().StringToStringP("env", "e", make(map[string]string), "environment variables for the vm")
 }
+
+func executeVM(
+	ctx context.Context, t deployer.TFPluginClient,
+	vm workloads.VM,
+	node uint32,
+	farm, disk, volume uint64,
+) error {
+	var diskMount workloads.Disk
+	if disk != 0 {
+		diskName := fmt.Sprintf("%sdisk", vm.Name)
+		diskMount = workloads.Disk{Name: diskName, SizeGB: disk}
+		vm.Mounts = []workloads.Mount{{Name: diskName, MountPoint: "/data"}}
+	}
+
+	var volumeMount workloads.Volume
+	if volume != 0 {
+		volumeName := fmt.Sprintf("%svolume", vm.Name)
+		volumeMount = workloads.Volume{Name: volumeName, SizeGB: volume}
+		vm.Mounts = append(vm.Mounts, workloads.Mount{Name: volumeName, MountPoint: "/volume"})
+	}
+
+	if node == 0 {
+		filter, ssd, rootfss := filters.BuildVMFilter(diskMount, volumeMount, farm, vm.MemoryMB, vm.RootfsSizeMB, vm.PublicIP, false)
+		nodes, err := deployer.FilterNodes(
+			ctx,
+			t,
+			filter,
+			ssd,
+			nil,
+			rootfss,
+		)
+		if err != nil {
+			return err
+		}
+
+		node = uint32(nodes[0].NodeID)
+	}
+
+	vm.NodeID = node
+	resVM, err := command.DeployVM(ctx, t, vm, diskMount, volumeMount)
+	if err != nil {
+		return err
+	}
+
+	if vm.PublicIP {
+		log.Info().Msgf("vm ipv4: %s", resVM.ComputedIP)
+	}
+	if vm.PublicIP6 {
+		log.Info().Msgf("vm ipv6: %s", resVM.ComputedIP6)
+	}
+	if vm.Planetary {
+		log.Info().Msgf("vm planetary ip: %s", resVM.PlanetaryIP)
+	}
+	if len(resVM.MyceliumIP) != 0 {
+		log.Info().Msgf("vm mycelium ip: %s", resVM.MyceliumIP)
+	}
+
+	return nil
+}
+
+func executeVMLight(
+	ctx context.Context, t deployer.TFPluginClient,
+	vm workloads.VMLight,
+	node uint32,
+	farm, disk, volume uint64,
+) error {
+	var diskMount workloads.Disk
+	if disk != 0 {
+		diskName := fmt.Sprintf("%sdisk", vm.Name)
+		diskMount = workloads.Disk{Name: diskName, SizeGB: disk}
+		vm.Mounts = []workloads.Mount{{Name: diskName, MountPoint: "/data"}}
+	}
+
+	var volumeMount workloads.Volume
+	if volume != 0 {
+		volumeName := fmt.Sprintf("%svolume", vm.Name)
+		volumeMount = workloads.Volume{Name: volumeName, SizeGB: volume}
+		vm.Mounts = append(vm.Mounts, workloads.Mount{Name: volumeName, MountPoint: "/volume"})
+	}
+
+	if node == 0 {
+		filter, ssd, rootfss := filters.BuildVMFilter(diskMount, volumeMount, farm, vm.MemoryMB, vm.RootfsSizeMB, false, true)
+		nodes, err := deployer.FilterNodes(
+			ctx,
+			t,
+			filter,
+			ssd,
+			nil,
+			rootfss,
+		)
+		if err != nil {
+			return errors.Wrap(errNoNodesFound, err.Error())
```

Comment:
deployer.FilterNodes already returns `ErrNoNodesMatchesResources` if no nodes found to match the required resources, I think you should use it instead.
check https://github.com/threefoldtech/tfgrid-sdk-go/blob/a5f890d80a127d88c4c2863bb5e91d6d9cedbd5d/grid-client/deployer/node_filter.go#L24